ste/overwritePrompter.go

```diff
@@ -39,23 +39,30 @@ type overwritePrompter struct {
 	savedResponse bool
 }
 
-func (o *overwritePrompter) shouldOverwrite(objectPath string) (shouldOverwrite bool) {
+func (o *overwritePrompter) ShouldOverwrite(objectPath string, objectType common.EntityType) (shouldOverwrite bool) {
 	// only one routine can ask the question or check the saved response at a time
 	o.lock.Lock()
 	defer o.lock.Unlock()
 
 	if o.shouldPromptUser {
-		shouldOverwrite = o.promptForConfirmation(objectPath)
+		shouldOverwrite = o.promptForConfirmation(objectPath, objectType)
 	} else {
 		shouldOverwrite = o.savedResponse
 	}
 
 	return
 }
 
-func (o *overwritePrompter) promptForConfirmation(objectPath string) (shouldDelete bool) {
-	answer := common.GetLifecycleMgr().Prompt(fmt.Sprintf("%s already exists at the destination. "+
-		"Do you wish to overwrite?", objectPath),
+func (o *overwritePrompter) promptForConfirmation(objectPath string, objectType common.EntityType) (shouldOverwrite bool) {
+	question := fmt.Sprintf("%s already exists at the destination. "+
+		"Do you wish to overwrite?", objectPath)
+
+	if objectType == common.EEntityType.Folder() {
+		question = fmt.Sprintf("Folder %s already exists at the destination. "+
```

Comment:
What about the case where the first prompt the user sees is about a folder.  And they answer yes to all, or no to all. Will they understand that their "all" answer, also applies to all file content?  I'm not confident that they will.  They may feel "I said yes to all on _folder properties_, but I didn't expect that to make it overwrite all _file content_"

So I wonder if this might help:

- Turn overwritePrompter.shouldPromptUser into an array, indexed by common.EntityType
- Then, everywhere we use it, the usage now looks something like this: `o.shouldPromptUser[objectType]`
- That way, and "all" answer on folders does not become also an "all" answer on files.

Questions to consider with that include:
- Should the messages "Confirmed. All future conflicts will be overwritten." and "o overwriting will happen from now onwards." be tweaked to include the entity type ("Folder Properties" vs "File")
- what if the first prompt is about a file overwrite.  It might be reasonable to expect a Yes to All or No to All to apply also to folders. (I.e. it seems intuitive for "all file content" to imply "and all folder properties".  But not, IMHO, the opposite way around)

Or is it overkill?  What do you think?

Reply:
It looks like:

> https://zemaintest.file.core.windows.net/testy/cmd/zt_credentialUtil_test.go already exists at the destination. Do you wish to overwrite? Please confirm with: [Y] Yes  [N] No  [A] Yes for all  [L] No for all a
INFO: Confirmed. All future file conflicts will be overwritten.
Folder https://zemaintest.file.core.windows.net/testy/cmd already exists at the destination. Do you wish to overwrite its properties? Please confirm with: [Y] Yes  [N] No  [A] Yes for all  [L] No for all l
INFO: No folder overwriting will happen from now onwards.